pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,776 @@
+	// state. Before you do it, please confirm that you understand the risks of this configuration.
+	//
+	// For supported flags, please see
+	// https://github.com/karmada-io/karmada/blob/master/cmd/webhook/app/options/options.go
+	// for details.
+	// +optional
+	ExtraArgs map[string]string `json:"extraArgs,omitempty"`
+}
+
+// CommonSettings describes the common settings of all karmada Components.
+type CommonSettings struct {
+	// ImageMeta allows to customize the image used for the component.
+	Image `json:",inline"`
+
+	// Number of desired pods. This is a pointer to distinguish between explicit
+	// zero and not specified. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Map of string keys and values that can be used to organize and categorize
+	// (scope and select) objects. May match selectors of replication controllers
+	// and services.
+	// More info: http://kubernetes.io/docs/user-guide/labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations is an unstructured key value map stored with a resource that may be
+	// set by external tools to store and retrieve arbitrary metadata. They are not
+	// queryable and should be preserved when modifying objects.
+	// More info: http://kubernetes.io/docs/user-guide/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Compute Resources required by this component.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type GlobalImage struct {
+	// karmada image repository and tag message.
+	// +optional
+	Karmada *Image `json:"karmada,omitempty"`
+
+	// Kubernetes image repository and tag message.
+	// +optional
+	Kubernetes *Image `json:"kubernetes,omitempty"`
+}
+
+// Image allows to customize the image used for components.
+type Image struct {
+	// ImageRepository sets the container registry to pull images from.
+	// if not set, the ImageRepository defined in KarmadaSpec will be used instead.
+	// +optional
+	ImageRepository string `json:"imageRepository,omitempty"`
+
+	// ImageTag allows to specify a tag for the image.
+	// In case this value is set, operator does not change automatically the version
+	// of the above components during upgrades.
+	// +optional
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+type HostCluster struct {
+	// The API endpoint of the member cluster. This can be a hostname,
+	// hostname:port, IP or IP:port.
+	// +optional
+	APIEndpoint string `json:"apiEndpoint,omitempty"`
+
+	// SecretRef represents the secret contains mandatory credentials to access the member cluster.
+	// The secret should hold credentials as follows:
+	// - secret.data.token
+	// - secret.data.caBundle
+	// +optional
+	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
```

Comment:
```suggestion
	// APIEndpoint is the API endpoint of the cluster where deploy Karmada
	// control plane on.
	// This can be a hostname, hostname:port, IP or IP:port.
	// +optional
	APIEndpoint string `json:"apiEndpoint,omitempty"`

	// SecretRef represents the secret contains mandatory credentials to
	// access the cluster.
	// The secret should hold credentials as follows:
	// - secret.data.token
	// - secret.data.caBundle
	// +optional
	SecretRef *LocalSecretReference `json:"secretRef,omitempty"`
```